src/rustc-driver-getting-diagnostics.md

```diff
@@ -0,0 +1,103 @@
+# Example: Getting diagnostic through `rustc_interface`
+
+`rustc_interface` allows you to intercept diagnostics that would otherwise be printed to stderr.
+
+## Getting diagnostics
+
+NOTE: For the example to compile, you will need to first run the following:
+
+    rustup component add rustc-dev
+
+To get diagnostics from the compiler, 
+configure `rustc_interface::Config` to output diagnostic to a buffer, 
+and run `TyCtxt.analysis`:
+
+```rust
+#![feature(rustc_private)]
```

Comment:
I mildly prefer to put the whole file in `src/examples/` as an actual rust file and only quote the relevant parts here? I'm not actually sure what the best way to handle this is...

cc @rust-lang/wg-rustc-dev-guide perhaps we should establish some sort of convention?

Reply:
It might also be worthwhile to run CI on the examples to make sure they continue to compile...

Reply:
Is it possible that we place files in `src/examples` and then display them in some way in the guide?. I imagine reading this in a tablet or something and being able to see the example right there in the section of the guide would be nice. @mark-i-m is that what you were suggesting? if so :+1:, I think that could be a nice convention.

Reply:
> It might also be worthwhile to run CI on the examples to make sure they continue to compile...

Good idea!